### PR TITLE
Svdesu/ch391/transparent nav bar

### PR DESCRIPTION
**Pull request by SVD6**
**Clubhouse Story:** https://app.clubhouse.io/chrono/story/391/transparent-nav-bar

**What does this PR solve?**
- When using an app I'm working on at work, I realized that a _translucent_ nav bar is actually not that bad at all, and theoretically it would be an easier solution than a fully _transparent_ one.
- I figured it would just be a flag in the styles, which for the most part it was. However turning the nav bar translucent meant it considered the space behind the nav bar as app-usable space. this meant onboarding buttons, circuit timer buttons, and some other stuff would end up behind the nav bar making them basically unusable and visually ugly.
- I created a function which can calculate the size of both the status bar and the nav bar. This then compensates the main layouts of these affected activities by adding enough margin to compensate for the status/nav bar. It works really well.
- Tested against Galaxy S8, Google Pixel, and Pixel 4XL. Tested against gesture navigation, and 3 button/2 button navigation.

**List the steps to take to replicate the behavior introduced by this PR**
- Best way to properly test this is with a device using 3-button navigation.
- Check all the screens throughout the app to see the behavior of the nav/status bar. The nav bar should be translucent, and the buttons/contents of the screen shouldn't be affected by it. No contents of the screen should be in the status bar as well (this happened at first, hence why I had to put margins to compensate for the status bar as well.

**List any known issues that this PR brings about**
- Thankfully, none that I'm aware of. As far as I can tell the app is the exact same with a translucent nav bar. The way the padding is being added to the top/bottom of layouts is pretty clean and doesn't require much code (like how nav bar transparency was being done before).

**What's something funny/stupid that happened when working on this story?**
- Nothing that I can think of :(

**Other information**

**Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines
- [x] All errors and warnings have been taken care of

### CODE REVIEW

File: app/src/main/java/ca/chronofit/chrono/util/helpers/UtilClass.kt [hunk 1]
```diff
@@ -35,4 +38,15 @@ fun formatTime(time: TimeObject, separator: String): String {
...
     text += time.milliseconds.toString() + time.tenMilliseconds.toString()
     return text
+}
+
+fun allocateForDeviceBars(res: Resources, view: ViewGroup) {
+    val navBar = res.getIdentifier("navigation_bar_height", "dimen", "android")
+    val statusBar = res.getIdentifier("status_bar_height", "dimen", "android")
+    val navBarSize = res.getDimensionPixelSize(navBar)
+    val statusBarSize = res.getDimensionPixelSize(statusBar)
+    println("debug: $navBarSize")
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/util/helpers/UtilClass.kt]: Do you want this in prod? If you do, log it, dont print.

**Comment by harigovind1998** [comment]
Pull in dev changes.

**Comment by harigovind1998** [comment]
image
image
Test results: 
It seems to be giving it a margin even when our nav bar isnt supposed to be there, namely in create circuit activity. Only seems to be happening when when dark mode is enabled (system default and our overwrite). 

Extra comments:
Also is it just me or does the gap at the top seem larger than usual?

**Review by harigovind1998 — CHANGES_REQUESTED** [review]
Please see test results above.

File: app/src/main/java/ca/chronofit/chrono/util/helpers/UtilClass.kt (hunk 1 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/util/helpers/UtilClass.kt]: Nah it was just there for dev purposes, got rid of it.

**Pull request closed without merging**